.github/workflows/ci.yml

```diff
@@ -0,0 +1,67 @@
+name: CI Workflow
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    services:
+      postgres:
+        image: postgres:14
+        env:
+          POSTGRES_HOST_AUTH_METHOD: trust
+          DB_USER: postgres
+          DB_PASSWORD: liduhgs4678^@RCGLSAYDuyio
+          DB_SERVER: db
+          DB_NAME: spotnet
+        ports:
+          - 5433:5432
+        options: >-
+          --health-cmd "pg_isready -U postgres" 
+          --health-interval 10s 
+          --health-timeout 5s 
+          --health-retries 5
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Set up Docker
+        uses: docker/setup-buildx-action@v2
+
+      - name: Install Docker Compose
+        run: |
+          sudo curl -L "https://github.com/docker/compose/releases/download/v2.20.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose
+          sudo chmod +x /usr/local/bin/docker-compose
+
+      - name: Create .env file
+        run: |
+          echo "DB_NAME=spotnet" >> .env
+          echo "DB_USER=postgres" >> .env
+          echo "DB_PASSWORD=liduhgs4678^@RCGLSAYDuyio" >> .env
+          echo "DB_HOST=db" >> .env
+      
+      - name: Start services
+        run: docker compose -f docker-compose.dev.yaml up -d --build 
+
+      - name: Wait for PostgreSQL
+        run: |
+          until docker-compose exec db pg_isready -U postgres; do
+            echo "Waiting for postgres..."
+            sleep 5
+          done
+
+      - name: Run tests
+        run: |
+          docker-compose exec backend pytest web_app/tests
```

Comment:
for running test cases, you don't need to connection to db

Reply:
Wdm? I didn't understand.